test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -429,6 +452,30 @@ func afterControlPlaneUpgradeTestHandler(ctx context.Context, c client.Client, n
 	}, intervals)
 }
 
+// beforeClusterDeleteHandler calls runtimeHookTestHandler with a blocking function which returns false if any of the
+// MachineDeployments associated with the Cluster are in a deleting state.
+func beforeClusterDeleteHandler(ctx context.Context, c client.Client, namespace, clusterName string, intervals []interface{}) {
+	hookName := "AfterControlPlaneUpgrade"
```

Comment:
```suggestion
	hookName := "BeforeClusterDelete"
```
Shouldn't this check for `BeforeClusterDelete`?